EventVisualisation/Workflow/include/EveWorkflow/EveWorkflowHelper.h

```diff
@@ -0,0 +1,42 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+///
+/// \file   EveWorkflowHelper.h
+/// \author julian.myrcha@cern.ch
+
+#ifndef ALICE_O2_EVENTVISUALISATION_WORKFLOW_EVEWORKFLOWHELPER_H
+#define ALICE_O2_EVENTVISUALISATION_WORKFLOW_EVEWORKFLOWHELPER_H
+
+#include "ReconstructionDataFormats/GlobalTrackID.h"
+#include "DataFormatsGlobalTracking/RecoContainer.h"
+#include "GPUDataTypes.h"
+//#include "GPUCommonMath.h"
+#include "EveWorkflow/EveConfiguration.h"
+
+namespace o2::event_visualisation
+{
+class EveWorkflowHelper
+{
+  using GID = o2::dataformats::GlobalTrackID;
+
+ public:
+  struct tmpDataContainer {
+    std::vector<o2::BaseCluster<float>> ITSClustersArray;
+    std::vector<int> tpcLinkITS, tpcLinkTRD, tpcLinkTOF;
+    std::vector<const o2::track::TrackParCov*> globalTracks;
+    std::vector<float> globalTrackTimes;
+  };
+  static std::shared_ptr<const tmpDataContainer> fillIOPtr(gpu::GPUTrackingInOutPointers& ioPtr, const o2::globaltracking::RecoContainer& recoCont, bool useMC, const CalibObjectsConst* calib = nullptr, GID::mask_t maskCl = GID::MASK_ALL, GID::mask_t maskTrk = GID::MASK_ALL, GID::mask_t maskMatch = GID::MASK_ALL);
```

Comment:
I don't understand what you do with the `gpu::GPUTrackingInOutPointers& ioPtr` here. If I understand correctly, you have copied my fillIOPtr function into your WorkflowHelper. I have a few remarks here:
- So first, I don't understand why you need the copying into the ioPtr at all. For the GPU this is only done since the GPU doesn't support all the data types. For the event display, you could just use the data directly without copying to pointers to a special structure beforehand.
- If you use my structure, why do you create a copy of fillIOPtr function? I would avoid code duplication if possible.
- If you want to fill the ioPtr by yourself, I would propose you also create your own ioPtr object. the gpu::GPUTrackingInOutPointers is no fixed API and it can change when the GPU code is developed further, so I don't like the dependency to the event display.

Reply:
I removed usage of ioPtr completely.